zipline/finance/controls.py

```diff
@@ -75,6 +78,16 @@ def fail(self, asset, amount, datetime, metadata=None):
                                       datetime=datetime,
                                       constraint=constraint)
 
+    def log(self, asset, amount, datetime, metadata=None):
```

Comment:
I think rather than making a new method here, we should just push the notion of `fail_on_violation` into the base class.